File: golang/devops_console/internal/domain/task/task.go
package task

import (
	"time"

	"github.com/google/uuid"
)

type TaskStatus string

const (
	TaskStatusPending            TaskStatus = "Pending"
	TaskStatusInProgress         TaskStatus = "InProgress"
	TaskStatusCompleted          TaskStatus = "Completed"
	TaskStatusFailed             TaskStatus = "Failed"
	TaskStatusScheduled          TaskStatus = "Scheduled"
	TaskStatusPendingValidation  TaskStatus = "PendingValidation"
)

type Form struct {
	ID     string            `json:"id"`
	Name   string            `json:"name"`
	Fields map[string]string `json:"fields"`
}

type Approval struct {
	ID           string     `json:"id"`
	UserID       string     `json:"user_id"`
	Approved     bool       `json:"approved"`
	ApprovalDate *time.Time `json:"approval_date,omitempty"`
}

type TaskExecuted struct {
	ID          string     `json:"id"`
	RunAt       time.Time  `json:"run_at"`
	WorkspaceID string     `json:"workspace_id"`
	Done        bool       `json:"done"`
	Status      TaskStatus `json:"status"`
}

type Task struct {
	ID           string                 `json:"id"`
	CreateAt     time.Time              `json:"create_at"`
	WorkspaceID  string                 `json:"workspace_id"`
	Name         string                 `json:"name"`
	TaskType     string                 `json:"task_type"`
	Technology   string                 `json:"technology"`
	WorkerType   *string                `json:"worker_type,omitempty"`
	Description  *string                `json:"description,omitempty"`
	ExtendedInfo *string                `json:"extended_info,omitempty"`
	Tags         []string               `json:"tags"`
	Forms        []Form                 `json:"forms"`
	Approvals    []Approval             `json:"approvals"`
	Metadata     map[string]string      `json:"metadata"`
	TasksExecuted []TaskExecuted        `json:"tasks_executed"`
}

type TaskCreate struct {
	WorkspaceID  string            `json:"workspace_id"`
	Name         string            `json:"name"`
	TaskType     string            `json:"task_type"`
	Technology   string            `json:"technology"`
	Description  *string           `json:"description,omitempty"`
	ExtendedInfo *string           `json:"extended_info,omitempty"`
	Tags         []string          `json:"tags"`
	Forms        []Form            `json:"forms"`
	Approvals    []Approval        `json:"approvals"`
	Metadata     map[string]string `json:"metadata"`
}

type TaskUpdate struct {
	Title        *string           `json:"title,omitempty"`
	TaskType     *string           `json:"task_type,omitempty"`
	Technology   *string           `json:"technology,omitempty"`
	Description  *string           `json:"description,omitempty"`
	ExtendedInfo *string           `json:"extended_info,omitempty"`
	Tags         *[]string         `json:"tags,omitempty"`
	Forms        *[]Form           `json:"forms,omitempty"`
	Approvals    *[]Approval       `json:"approvals,omitempty"`
	Metadata     map[string]string `json:"metadata,omitempty"`
	TasksExecuted *[]TaskExecuted  `json:"tasks_executed,omitempty"`
}

type TaskRepository interface {
	Create(task TaskCreate) (*Task, error)
	GetAll() ([]*Task, error)
	GetByID(taskID string) (*Task, error)
	Update(taskID string, taskUpdate TaskUpdate) (*Task, error)
	Delete(taskID string) error
}

File: backend/src/domain/task/task.py
from pydantic import BaseModel, Field
from typing import Optional, List, Dict
from datetime import datetime
from enum import Enum
import uuid

from src.infrastructure.workers.worker_types import WorkerType

class TaskStatus(str, Enum):
    PENDING = 'Pending'
    IN_PROGRESS = 'InProgress'
    COMPLETED = 'Completed'
    FAILED = 'Failed'
    SCHEDULED = 'Scheduled'
    PENDING_VALIDATION = 'PendingValidation'

class Form(BaseModel):
    id: str
    name: str
    fields: Dict[str, str]

class Approval(BaseModel):
    id: str
    user_id: str
    approved: bool
    approval_date: Optional[datetime]

class TaskExecuted(BaseModel):
    id: str
    run_at: datetime
    workspace_id: str
    done: bool
    status: TaskStatus

class Task(BaseModel):
    id: str = Field(default_factory=lambda: str(uuid.uuid4()))
    create_at: datetime = Field(default_factory=datetime.utcnow)
    workspace_id: str
    name: str
    task_type: str
    technology: str
    worker_type: Optional[WorkerType] = None
    description: Optional[str] = None
    extended_info: Optional[str] = None
    tags: List[str] = []
    forms: List[Form] = []
    approvals: List[Approval] = []
    metadata: Dict[str, str] = {}
    tasks_executed: List[TaskExecuted] = []

class TaskCreate(BaseModel):
    workspace_id: str
    name: str
    task_type: str
    technology: str
    description: Optional[str] = None
    extended_info: Optional[str] = None
    tags: List[str] = []
    forms: List[Form] = []
    approvals: List[Approval] = []
    metadata: Dict[str, str] = {}

class TaskUpdate(BaseModel):
    title: Optional[str] = None
    task_type: Optional[str] = None
    technology: Optional[str] = None
    description: Optional[str] = None
    extended_info: Optional[str] = None
    tags: Optional[List[str]] = None
    forms: Optional[List[Form]] = None
    approvals: Optional[List[Approval]] = None
    metadata: Optional[Dict[str, str]] = None
    tasks_executed: Optional[List[TaskExecuted]] = None
